modules/javafx.graphics/src/main/native-glass/gtk/GlassApplication.cpp

```diff
@@ -112,7 +113,20 @@ JNIEXPORT jint JNICALL Java_com_sun_glass_ui_gtk_GtkApplication__1initGTK
     gdk_threads_enter();
     gtk_init(NULL, NULL);
 
-    return JNI_TRUE;
+    // Major version is checked before loading
+    if (version == 3) {
+        if(gtk_check_version(version, GTK_3_MIN_MINOR_VERSION, GTK_3_MIN_MICRO_VERSION)) {
+            char message[100];
+            std::sprintf(message, "Minimum GTK version required is %d.%d.%d. System has %d.%d.%d.",
```

Comment:
Please change this to `snprintf`, which take the length of the array. We should not be using `sprintf` directly as that can lead to buffer overflow.

Reply:
Done.